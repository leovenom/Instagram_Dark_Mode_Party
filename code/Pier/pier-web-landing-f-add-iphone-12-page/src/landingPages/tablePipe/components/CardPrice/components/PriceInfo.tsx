import * as React from "react";

import Text from "ui/Text";
import data from "../data";

import {
  getArrowsUiOrientation,
  getMonthlyValue,
  getInsuredValue,
} from "../../utils";
import { CardRow, CardRowItem } from "../styles";

interface PriceInfo {
  smartphoneSelected: {
    plansValues: object;
    name: string;
    memory: string;
    model: {
      name: string;
    };
  };
  variant: string;
}

const PriceInfo: React.FC<PriceInfo> = ({ smartphoneSelected, variant }) => {
  const arrowsOrientation = getArrowsUiOrientation({
    smartphoneSelected,
    variant,
  });

  return (
    <>
      <CardRow>
        <CardRowItem>
          <Text variant="bodySmall" color="gray600">
            {data?.plan}
          </Text>
        </CardRowItem>
        <CardRowItem>
          <Text
            variant="bodySmallest"
            bold
            transform="uppercase"
            color="primary400"
          >
            {data?.economic}
          </Text>
        </CardRowItem>
        <CardRowItem>
          <Text
            variant="bodySmallest"
            bold
            transform="uppercase"
            color="primary400"
          >
            {data?.premium}
          </Text>
        </CardRowItem>
      </CardRow>

      <CardRow>
        <CardRowItem>
          <Text variant="bodySmall" color="gray600">
            {data?.monthlyPayment}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.monthly?.economic}>
          <Text variant="bodySmall" color="text">
            {getMonthlyValue("economic", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.monthly?.premium}>
          <Text variant="bodySmall" color="text">
            {getMonthlyValue("premium", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
      </CardRow>

      <CardRow>
        <CardRowItem>
          <Text variant="bodySmall" color="gray600">
            {data?.totalInsuranceValue}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.insuredValue?.economic}>
          <Text variant="bodySmall" color="text">
            {getInsuredValue("economic", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.insuredValue?.premium}>
          <Text variant="bodySmall" color="text">
            {getInsuredValue("premium", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
      </CardRow>

      <Text variant="body" align="center">
        {data.insuranceValueUsed}{" "}
        <strong>
          {getInsuredValue("premium", smartphoneSelected)?.formatted}
        </strong>
      </Text>
    </>
  );
};

export default PriceInfo;
