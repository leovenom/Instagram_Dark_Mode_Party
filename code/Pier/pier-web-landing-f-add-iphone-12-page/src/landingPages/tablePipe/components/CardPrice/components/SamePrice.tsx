import Text from "ui/Text";
import data from "../data";

import { ImgWoman } from "../styles";

const SamePrice = () => (
  <>
    <ImgWoman
      src={data.samePrice.imageSrc}
      alt={data.samePrice.imageDescription}
    />
    <Text
      variant="tag"
      bold
      transform="uppercase"
      align="center"
      color="primary400"
      mb={10}
    >
      {data.samePrice.title}
    </Text>
    <Text variant="bodySmall" align="center" color="gray600">
      {data.samePrice.description}
    </Text>
  </>
);

export default SamePrice;
