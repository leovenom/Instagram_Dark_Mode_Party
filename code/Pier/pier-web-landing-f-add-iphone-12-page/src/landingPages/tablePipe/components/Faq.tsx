import Grid from "ui/Grid";
import Collapse from "ui/Collapse";
import Section from "landingPages/sharedSections/Section";
import { smartphoneTracker } from "helpers/mixpanelTracker";

import {
  StyledTitleGridItem,
  StyledText,
  StyledCollapseGridItem,
  StyledLink,
} from "../styles";

import data from "../data";

const Faq = () => (
  <Section bg="white" id="section-faq" pt={0} pb={110}>
    <Grid>
      <StyledTitleGridItem m={2} t={8} d={4}>
        <StyledText mb={10} variant="sectionSecondary">
          {data.faq.title}
        </StyledText>
        <StyledText variant="body" color="gray">
          {data.faq.subtitle}
        </StyledText>
      </StyledTitleGridItem>
      <StyledCollapseGridItem m={2} t={8} d={8}>
        {data.faq.questions.map(({ question, answer }) => (
          <Collapse key={question}>
            <Collapse.Title>{question}</Collapse.Title>
            <Collapse.Content>{answer}</Collapse.Content>
          </Collapse>
        ))}
        <StyledLink
          withArrow
          href="https://ajuda.pier.digital/pt-BR/articles/2887266-atualizacao-da-tabela-pipe"
          target="_blank"
          rel="noopener"
          onClick={() =>
            smartphoneTracker.trackButton("Tabela Pipe See More Answers")
          }
        >
          {data.faq.link}
        </StyledLink>
      </StyledCollapseGridItem>
    </Grid>
  </Section>
);

export default Faq;
