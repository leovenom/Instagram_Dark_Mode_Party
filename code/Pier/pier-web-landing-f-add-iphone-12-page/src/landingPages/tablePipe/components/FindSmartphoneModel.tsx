import * as React from "react";

import Text from "ui/Text";
import Button from "ui/Button";

import { smartphoneTracker } from "helpers/mixpanelTracker";
import data from "../data";

interface FindSmartphoneModelProps {
  smartphoneSelected: {
    plansValues: object;
    name: string;
    memory: string;
    model: {
      name: string;
    };
  };
  toggleSmartphoneSelector: () => void;
}

const FindSmartphoneModel: React.FC<FindSmartphoneModelProps> = ({
  toggleSmartphoneSelector,
  smartphoneSelected,
}) => {
  if (smartphoneSelected) {
    return null;
  }
  const handleOpenModalSmartphoneModels = () => {
    smartphoneTracker.trackButton("Tabela Pipe Prices");
    toggleSmartphoneSelector();
  };

  return (
    <>
      <Text variant="section" align="center" mb={24}>
        {data.findSmartphoneModel.title}
      </Text>
      <Text variant="body" align="center" mb={40}>
        {data.findSmartphoneModel.subtitle}
      </Text>
      <Button size="small" onClick={handleOpenModalSmartphoneModels}>
        {data.findSmartphoneModel.cta}
      </Button>
    </>
  );
};

export default FindSmartphoneModel;
