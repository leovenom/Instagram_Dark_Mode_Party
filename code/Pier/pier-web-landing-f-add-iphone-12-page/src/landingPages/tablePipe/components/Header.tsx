import HeroCenteredContent from "landingPages/sharedSections/heroSections/HeroCenteredContent";
import data from "../data";

const Header = () => (
  <HeroCenteredContent
    bg="orange400"
    title={data.header.title}
    subtitle={data.header.subtitle}
    imageSrc={data.header.imageSrc}
    imageDescription={data.header.imageDescription}
    withoutTabs
  />
);

export default Header;
