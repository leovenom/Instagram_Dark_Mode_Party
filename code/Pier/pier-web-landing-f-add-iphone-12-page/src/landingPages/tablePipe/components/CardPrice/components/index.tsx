import PriceInfo from "./PriceInfo";
import SamePrice from "./SamePrice";

export { PriceInfo, SamePrice };
