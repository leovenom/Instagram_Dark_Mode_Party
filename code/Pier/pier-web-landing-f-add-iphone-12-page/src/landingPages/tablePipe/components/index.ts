import Header from "./Header";
import Instructions from "./Instructions";
import FindSmartphoneModel from "./FindSmartphoneModel";
import Faq from "./Faq";
import ShowPrices from "./ShowPrices";

export { Header, Instructions, FindSmartphoneModel, Faq, ShowPrices };
