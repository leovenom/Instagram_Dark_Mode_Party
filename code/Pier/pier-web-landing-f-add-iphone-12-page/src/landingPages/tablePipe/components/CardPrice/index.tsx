import CardPrice from "./CardPrice";

export default CardPrice;
