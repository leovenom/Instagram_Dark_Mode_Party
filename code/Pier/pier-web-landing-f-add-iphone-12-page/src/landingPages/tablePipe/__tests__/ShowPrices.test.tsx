import { fireEvent } from "@testing-library/react";

import { smartphoneTracker } from "helpers/mixpanelTracker";
import { renderWithTheme } from "helpers/testUtils";
import { ShowPrices } from "../components";
import {
  getInsuredValue,
  getMonthlyValue,
  getNewPriceOfPlan,
} from "../components/utils";

jest.mock("helpers/mixpanelTracker");
const toggleSmartphoneSelector = jest.fn();

const renderComponent = ({ smartphoneSelected }) =>
  renderWithTheme(
    <ShowPrices
      toggleSmartphoneSelector={toggleSmartphoneSelector}
      smartphoneSelected={smartphoneSelected}
    />
  );

const selectedSmartphone = {
  deviceOs: "ios",
  manufacturer: undefined,
  memory: "16GB",
  model: { memories: ["16GB", "32GB", "64GB", "128GB"], name: "iPhone 6S" },
  plansValues: {
    premium: { insuredValue: 712.0, price: 13.9 },
    economic: { insuredValue: 890.0, price: 19.1 },
  },
  serie: "Apple",
};

describe("<ShowPrices />", () => {
  it("Should render without errors", () => {
    const { container } = renderComponent({
      smartphoneSelected: selectedSmartphone,
    });
    expect(container).toMatchSnapshot();
  });

  describe("When the button 'Selecione outro modelo' is clicked", () => {
    it("Should trigger a track button event", () => {
      const { getByText } = renderComponent({
        smartphoneSelected: selectedSmartphone,
      });

      fireEvent.click(getByText("Selecione outro modelo"));

      expect(smartphoneTracker.trackButton).toHaveBeenCalledWith(
        "Tabela Pipe Select Other Model"
      );
    });

    it("Should invoke the toggleSmartphoneSelector fn", () => {
      const { getByText } = renderComponent({
        smartphoneSelected: selectedSmartphone,
      });

      fireEvent.click(getByText("Selecione outro modelo"));

      expect(toggleSmartphoneSelector.mock.calls.length).toBe(1);
    });
  });

  describe("When the smartphoneSelected object is not empty", () => {
    it("Should render the component", () => {
      const { getByText } = renderComponent({
        smartphoneSelected: selectedSmartphone,
      });

      getByText("Valores para um");
      getByText("iPhone 6S 16GB");
      getByText("Selecione outro modelo");
    });

    it("Should render the component with the correct smartphone model", () => {
      const { getByText } = renderComponent({
        smartphoneSelected: selectedSmartphone,
      });

      // iPhone 6S 16GB
      getByText(
        `${selectedSmartphone.model.name} ${selectedSmartphone.memory}`
      );
    });

    it("Should render the component with the correct update dates in cards", () => {
      const { getByText } = renderComponent({
        smartphoneSelected: selectedSmartphone,
      });

      // New Card
      getByText("A partir de 23/11/2020");

      // Old Card
      getByText("Anterior a 23/11/2020");
    });

    it("Should trigger a track screen event", () => {
      renderComponent({ smartphoneSelected: selectedSmartphone });

      expect(smartphoneTracker.trackScreen).toHaveBeenCalledWith(
        "Tabela Pipe Plans"
      );
    });
  });

  describe("When the smartphone model selected has a new plan price", () => {
    const selectedSmartphone = {
      deviceOs: "ios",
      manufacturer: undefined,
      memory: "512GB",
      model: {
        memories: ["64GB", "256GB", "512GB"],
        name: "iPhone 11 Pro Max",
      },
      plansValues: {
        premium: { insuredValue: 6144.0, price: 119.9 },
        economic: { insuredValue: 7680.0, price: 164.8 },
      },
      serie: "Apple",
    };

    const newPlanPrice = getNewPriceOfPlan({
      smartphoneSelected: selectedSmartphone,
    });

    it("Should render the component with the old plan card and new plan card", () => {
      const { getByText, getAllByText } = renderComponent({
        smartphoneSelected: selectedSmartphone,
      });

      // Plans Values
      getByText(getMonthlyValue("economic", newPlanPrice).formatted);
      getByText(getMonthlyValue("economic", selectedSmartphone).formatted);

      getByText(getMonthlyValue("premium", newPlanPrice).formatted);
      getByText(getMonthlyValue("premium", selectedSmartphone).formatted);

      getAllByText(getInsuredValue("economic", newPlanPrice).formatted);
      getAllByText(getInsuredValue("economic", selectedSmartphone).formatted);

      getAllByText(getInsuredValue("premium", newPlanPrice).formatted);
      getAllByText(getInsuredValue("premium", selectedSmartphone).formatted);
    });
  });

  describe("When the smartphone model selected doesn't have a new plan price", () => {
    it("Should render the component with 'no updates info' card", () => {
      const { getByText } = renderComponent({
        smartphoneSelected: selectedSmartphone,
      });

      getByText("Sem reajustes");
      getByText("Nada mudou para você nessa atualização!");
    });
  });

  describe("When the smartphoneSelected object is empty", () => {
    it("Should not render the component", () => {
      const { queryByText } = renderComponent({ smartphoneSelected: null });

      expect(queryByText("Valores para um")).toBeFalsy();
      expect(queryByText("iPhone 6S 16GB")).toBeFalsy();
      expect(queryByText("Selecione outro modelo")).toBeFalsy();
    });

    it("Shouldn't trigger a track screen event", () => {
      renderComponent({ smartphoneSelected: null });

      expect(smartphoneTracker.trackScreen).not.toHaveBeenCalled();
    });
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });
});
