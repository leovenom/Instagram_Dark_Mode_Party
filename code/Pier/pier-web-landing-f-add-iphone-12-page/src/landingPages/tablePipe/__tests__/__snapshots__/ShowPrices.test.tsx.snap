// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowPrices /> Should render without errors 1`] = `
<div>
  <div
    class="Grid__StyledGrid-sc-148v7sh-0 klckme"
  >
    <div
      class="GridItem__StyledGridItem-rojpzy-0 jmIaJh"
      d="12"
    >
      <img
        alt="Desenho de um Smartphone"
        class="styles__SmartphoneImg-sc-18zimf4-6 eOxcPy"
        src="/static_assets/images/rebranding/tilted-smartphone.svg"
      />
      <p
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 bmrwnc"
      >
        Valores para um
      </p>
      <h2
        class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 iKpMtv"
      >
        iPhone 6S
         
        16GB
      </h2>
    </div>
    <div
      class="GridItem__StyledGridItem-rojpzy-0 kFuxWN"
      d="12"
    >
      <div
        class="Grid__StyledGrid-sc-148v7sh-0 klckme"
      >
        <div
          class="GridItem__StyledGridItem-rojpzy-0 fARTXv"
          d="6"
        >
          <div
            class="styles__CardContainer-v1rd0i-0 jPFCtx"
          >
            <div
              class="styles__CardHeader-v1rd0i-1 iDzLUt"
            >
              <h5
                class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 flYYBQ"
                color="primary"
              >
                Antes da atualização
              </h5>
              <p
                class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 kJMJxo"
                color="gray600"
              >
                Anterior a 23/11/2020
              </p>
            </div>
            <div
              class="styles__CardContent-v1rd0i-2 bKyhLl"
            >
              <div
                class="styles__CardRow-v1rd0i-3 kZOViC"
              >
                <div
                  class="styles__CardRowItem-v1rd0i-4 leoyfg"
                >
                  <p
                    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 kJMJxo"
                    color="gray600"
                  >
                    Plano
                  </p>
                </div>
                <div
                  class="styles__CardRowItem-v1rd0i-4 leoyfg"
                >
                  <p
                    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 cDGqPZ"
                    color="primary400"
                    transform="uppercase"
                  >
                    Econômico
                  </p>
                </div>
                <div
                  class="styles__CardRowItem-v1rd0i-4 leoyfg"
                >
                  <p
                    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 cDGqPZ"
                    color="primary400"
                    transform="uppercase"
                  >
                    Premium
                  </p>
                </div>
              </div>
              <div
                class="styles__CardRow-v1rd0i-3 kZOViC"
              >
                <div
                  class="styles__CardRowItem-v1rd0i-4 leoyfg"
                >
                  <p
                    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 kJMJxo"
                    color="gray600"
                  >
                    Mensalidade
                  </p>
                </div>
                <div
                  class="styles__CardRowItem-v1rd0i-4 leoyfg"
                >
                  <p
                    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 hEfGWY"
                    color="text"
                  >
                    R$ 19,10
                  </p>
                </div>
                <div
                  class="styles__CardRowItem-v1rd0i-4 leoyfg"
                >
                  <p
                    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 hEfGWY"
                    color="text"
                  >
                    R$ 13,90
                  </p>
                </div>
              </div>
              <div
                class="styles__CardRow-v1rd0i-3 kZOViC"
              >
                <div
                  class="styles__CardRowItem-v1rd0i-4 leoyfg"
                >
                  <p
                    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 kJMJxo"
                    color="gray600"
                  >
                    Valor do reembolso
                  </p>
                </div>
                <div
                  class="styles__CardRowItem-v1rd0i-4 leoyfg"
                >
                  <p
                    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 hEfGWY"
                    color="text"
                  >
                    R$ 890,00
                  </p>
                </div>
                <div
                  class="styles__CardRowItem-v1rd0i-4 leoyfg"
                >
                  <p
                    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 hEfGWY"
                    color="text"
                  >
                    R$ 712,00
                  </p>
                </div>
              </div>
              <p
                class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 difqfl"
              >
                Valor de um semi-novo: 
                 
                <strong>
                  R$ 712,00
                </strong>
              </p>
            </div>
          </div>
        </div>
        <div
          class="GridItem__StyledGridItem-rojpzy-0 hMyuob"
          d="6"
        >
          <div
            class="styles__CardContainer-v1rd0i-0 jPFCtx"
          >
            <div
              class="styles__CardHeader-v1rd0i-1 iDzLUt"
            >
              <h5
                class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 flYYBQ"
                color="primary"
              >
                Após a atualização
              </h5>
              <p
                class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 kJMJxo"
                color="gray600"
              >
                A partir de 23/11/2020
              </p>
            </div>
            <div
              class="styles__CardContent-v1rd0i-2 bKyhLl"
            >
              <img
                alt="Pessoa pensativa ao lado de um celular indicando queda de preço"
                class="styles__ImgWoman-v1rd0i-5 MJvyN"
                src="/static_assets/images/rebranding/woman-with-smartphone.svg"
              />
              <p
                class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 gLpnGR"
                color="primary400"
                transform="uppercase"
              >
                Sem reajustes
              </p>
              <p
                class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 eQVYHj"
                color="gray600"
              >
                Nada mudou para você nessa atualização!
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="GridItem__StyledGridItem-rojpzy-0 jmIaJh"
      d="12"
    >
      <a
        class="ExternalLink__StyledLink-lmn7o3-0 kVBygc styles__StyledLink-sc-18zimf4-5 emsBSY"
        rel="noopener noreferrer"
        target="_blank"
      >
        Selecione outro modelo
         
        <svg
          fill="none"
          height="8"
          viewBox="0 0 14 8"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 1L7 7L1 1"
            stroke="#24263B"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
